### PR TITLE
ceph: use a different port for rgw on sdn

### PR DESCRIPTION
**Pull request by leseb**
**Description of your changes:**

When running on SDN, the container is not privileged and the network
stack is not exposed so running the gw on port 80 will not work (not
enough privileged).

Closes: https://github.com/rook/rook/issues/5106
Signed-off-by: Sébastien Han <seb@redhat.com>

**Which issue is resolved by this Pull Request:**
Resolves https://github.com/rook/rook/issues/5106

**Checklist:**

- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.
- [ ] Comments have been added or updated based on the standards set in [CONTRIBUTING.md](https://github.com/rook/rook/blob/master/CONTRIBUTING.md#comments)
- [ ] Add the flag for skipping the CI if this PR does not require a build. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for more details.

[test ceph]

### CODE REVIEW

File: pkg/operator/ceph/object/spec.go [hunk 1]
```diff
@@ -124,6 +124,12 @@ func (c *clusterConfig) makeChownInitContainer(rgwConfig *rgwConfig) v1.Containe
 }
 
 func (c *clusterConfig) makeDaemonContainer(rgwConfig *rgwConfig) v1.Container {
+	// Liveness probe port
```
Comment (travisn) [inline pkg/operator/ceph/object/spec.go]: nit: liveness probe?

File: pkg/operator/ceph/object/spec.go [hunk 2]
```diff
@@ -124,6 +124,12 @@ func (c *clusterConfig) makeChownInitContainer(rgwConfig *rgwConfig) v1.Containe
...
+	port := intstr.FromInt(int(rgwPortInternalPort))
+	if c.clusterSpec.Network.IsHost() {
+		port = intstr.FromInt(int(c.store.Spec.Gateway.Port))
```
Comment (travisn) [inline pkg/operator/ceph/object/spec.go]: per discussion we need to understand if the liveness probe works with https (`SecurePort`)

Comment (leseb) [reply pkg/operator/ceph/object/spec.go]: it does! with `URISchemeHTTPS`, will send a fix that too

File: pkg/operator/ceph/object/config.go [hunk 3]
```diff
@@ -50,6 +51,9 @@ func (c *clusterConfig) portString() string {
 
 	port := c.store.Spec.Gateway.Port
 	if port != 0 {
```
Comment (travisn) [inline pkg/operator/ceph/object/config.go]: Could port selection here and for the liveness probe share the helper method `generateLiveProbePort()`? Seems like both are computing the same thing separately, or is the difference needed?

**Pull request merged**